src/program/mod.rs

```diff
@@ -130,15 +130,10 @@ impl Program {
         &'a self,
         instruction: &'a Instruction,
         include_blocked: bool,
-    ) -> Option<Vec<&'a FrameIdentifier>> {
+    ) -> Option<HashSet<&'a FrameIdentifier>> {
         instruction
             .get_frame_match_condition(include_blocked)
-            .map(|condition| {
-                self.frames
-                    .get_matching_keys(condition)
-                    .into_iter()
-                    .collect()
-            })
+            .map(|condition| self.frames.get_matching_keys(condition))
```

Comment:
How come this doesn't need a `.collect()`? 🤔

Reply:
Because `get_matching_keys` already returns a `HashSet`?

That said, you might be able to shorten this to `.map(self.frames.get_matching_keys)`.

Reply:
Compiler doesn't like passing that function to `map`, although I'm not sure why.